docs/standard/native-interop/abi-support.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Native interoperability ABI support - .NET
+description: Defines the current support for interoperability with various ABIs.
+ms.date: 10/25/2024
+---
+# Native interoperability ABI support
+
+[Application Binary Interface](https://wikipedia.org/wiki/Application_binary_interface) (ABI) is the interface which run times and operating systems express low-level binary details. This can include calling conventions (that is, how parameters are passed and results returned), exception handling, as well as symbol mangling. The below list contains the names of languages, run times and even general technologies since these names tend to represent what users will be working with when looking for guidance on interoperability.
+
+## Support table
+
+|   | Support | APIs |
+|---|---------|------|
+| **C**                  | ✔️ | <xref:System.Runtime.InteropServices.LibraryImportAttribute> provides source generated support in .NET 7+. Use <xref:System.Runtime.InteropServices.DllImportAttribute> when targeting earlier .NET versions. Refer to [Interop best practices](./best-practices.md) for additional guidance. |
+| **C++**                | ❌ | [Guidance](#cpp) |
+| **COM and `IUnknown`** | ✔️ | In .NET 5+, low-level, cross-platform, `IUnknown` lifetime support is provided by <xref:System.Runtime.InteropServices.ComWrappers>. In .NET 8+, <xref:System.Runtime.InteropServices.Marshalling.GeneratedComInterfaceAttribute> and <xref:System.Runtime.InteropServices.Marshalling.GeneratedComClassAttribute> provide source generated C# projections. The built-in COM interop system is limited to the Windows platforms and required on versions prior to .NET 5. |
+| **Swift**              | ✔️ | For .NET 9+, <xref:System.Runtime.InteropServices.Swift> namespace.      |
+| **Objective-C**        | ✔️ | For .NET 8+, <xref:System.Runtime.InteropServices.ObjectiveC> namespace. |
```

Comment:
Also mention the existing tooling for OS APIs? Also maybe mention ObjectiveSharpie for user-defined APIs?